lib/Sema/SemaChecking.cpp

```diff
@@ -11582,46 +11638,58 @@ bool Sema::DiagnoseCheckedDecl(const ValueDecl *Decl, SourceLocation UseLoc) {
   }
   else if (const VarDecl *Var = dyn_cast<VarDecl>(Decl)) {
     TargetDecl = Var;
-    DeclKind = 2; // decl var
+    DeclKind = Var->isLocalVarDecl() ? 2 : 3; // decl var
```

Comment:
Where are these constants coming from? Are they for the error reporting? Do we need a local enum?

Reply:
Replaced magic constants with an enum.